### PR TITLE
Good riddance

### LINKED ISSUE #3108: Refactor and redesign Tags Admin
The code that powers Tags admin UI has been in the past used on Tags dashboards as well. Thus, it is quite complex for the current use case and should be replaced with something simpler.

We should also replace the Tags UI widget.

### PR DESCRIPTION
**Pull request by mathjazz**
Fix #3108.

1. First and foremost, this patch replaces the unusual widget for selecting tag resources in Pontoon Admin with the multiple item selector that is more commonly used in Pontoon (see [Add new Tag resource management UI](https://github.com/mozilla/pontoon/pull/3367/commits/66fe09ef1b9b734ef21b1593dd23f908a4cf779b)).

2. Secondly, the last bits of the old [Tags Django code](https://github.com/mozilla/pontoon/commit/d36f01786ff531d259e4d13e4e3d8037243aa9a6) are now finally removed. It should now be much simpler to fix issues like #2122 and #2131.

3. Finally, the odd tag-admin frontend, which was only powering the old widget for selecting tag resources in Pontoon Admin, is now removed (see [Remove Tag Admin Frontend code](https://github.com/mozilla/pontoon/pull/3367/commits/81fe637c6de91a4d6d517f2c0f0692a46b7a7021) and [Remove Tag Admin Frontend environment](https://github.com/mozilla/pontoon/pull/3367/commits/4a5012a58d1ebdd3dc8b6676a28a0882e0e4f4e1)).

In total, the same (actually, I think it's much better) functionality uses some 95% smaller footprint:
Screenshot 2024-09-22 at 16 23 43

The patch was tested locally in Docker.

It should also be tested on stage before we can proceed:
https://mozilla-pontoon-staging.herokuapp.com/admin/projects/firefox/

Also included are a few smaller Pontoon Admin improvements:
* [Unify dropbox style with the rest of the UI](https://github.com/mozilla/pontoon/commit/9e65ccedf7154df17fe7a8a9f34b0f6d37419df5)
* [Use native date selector widget for target date](https://github.com/mozilla/pontoon/commit/91296d3acb5aa1bec492895cdcc92babc2d00419)
* [Fix large margin in the external resources form](https://github.com/mozilla/pontoon/commit/0b36fd030e2710f1042bcaba1b749a93f1ec45bf)

A few additional changes would be nice to make in Pontoon Admin. At least one of them would clash with #3353, so I've filed #3368 for that.

### CODE REVIEW

**Comment by flodolo** [comment]
> This patch should be tested on stage before we can proceed: [mozilla-pontoon-staging.herokuapp.com/admin/projects/firefox](https://mozilla-pontoon-staging.herokuapp.com/admin/projects/firefox/)

I assume this isn't on stage anymore at this point?

**Comment by mathjazz** [comment]
> I assume this isn't on stage anymore at this point?

It will be again in a few mintues, just finishing deploying #3378, #3353 and #3367.

**Comment by flodolo** [comment]
> [Unify dropbox style with the rest of the UI](https://github.com/mozilla/pontoon/commit/9e65ccedf7154df17fe7a8a9f34b0f6d37419df5)

I don't remember if we have docs with screenshots of the admin UI. In case, they will need to be redone (I like the more consistent look).

* Available-Chosen: should we stick to Linked instead of Chosen?
* Unlike in the locales section, there is no visual representation (< or > on hover) that clicking will move an item in the other column. Is that by design?

Something is broken though: I moved one file from `mobile` into one tag, and now the list of resources doesn't load anymore.
https://mozilla-pontoon-staging.herokuapp.com/ace/firefox/

**Comment by mathjazz** [comment]
> I don't remember if we have docs with screenshots of the admin UI. In case, they will need to be redone (I like the more consistent look).

I only found this: https://mozilla-l10n.github.io/documentation/tools/pontoon/adding_new_project.html

The screenshot only contains the tag management part of the Admin UI, so we can update it (and the corresponding text) as soon as this PR is merged.

> Available-Chosen: should we stick to Linked instead of Chosen?

This is currently hardcoded in the widget. Due to the clash with #3353, we could implement it as part of #3368 (or a separate issue).

> Unlike in the locales section, there is no visual representation (< or > on hover) that clicking will move an item in the other column. Is that by design?

The locale widget is specific, because an arrow is a must in the middle column. In other places where the widget is used, we don't use an arrow in order to save space, see user settings or Messaging center. 

We could still add an arrow, but for the reason mentioned above, that would need to be in #3368 (or a separate issue).

> Something is broken though: I moved one file from `mobile` into one tag, and now the list of resources doesn't load anymore.

Nice catch! Looking into it.

**Comment by mathjazz** [comment]
> Nice catch! Looking into it.

Update: Clicking on a resource unsets the tag priority, and the dashboard breaks in case tags don't have a priority. Interestingly though I can't reproduce that locally yet.

We should also make tag priority a required field (because even on `main`, if you remove priority from tag by mistake, the dashboard will break).

**Comment by mathjazz** [comment]
> Update: Clicking on a resource unsets the tag priority, and the dashboard breaks in case tags don't have a priority. Interestingly though I can't reproduce that locally yet.

Turns out this is a bug in https://github.com/mozilla/pontoon/pull/3353, which is also deployed to stage. 🤣

**Comment by mathjazz** [comment]
@flodolo Updated the stage with fixes made in https://github.com/mozilla/pontoon/pull/3353.

**Comment by flodolo** [comment]
The admin area looks good on stage. Is there anything else that needs testing?

**Comment by mathjazz** [comment]
> The admin area looks good on stage. Is there anything else that needs testing?

Thanks. Mostly Admin, but also the Tag dashboards.

**Review by flodolo — APPROVED** [review]
Both admin area and tag dashboard look good to me.

**Review by flodolo — APPROVED** [review]

File: pontoon/settings/base.py [hunk 1]
```diff
@@ -422,9 +421,9 @@ def _default_from_email():
     "admin_project": {
         "source_filenames": (
             "css/double_list_selector.css",
+            "css/multiple_item_selector.css",
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/settings/base.py:424]: Duplication of?

File: pontoon/base/models/resource.py [hunk 2]
```diff
@@ -82,6 +82,10 @@ def allows_empty_translations(self):
         """Return True if this resource allows empty translations."""
         return self.format in self.EMPTY_TRANSLATION_FORMATS
 
+    @property
+    def name(self):
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/base/models/resource.py:86]: We already have `Resource.path`. We need `name` in https://github.com/mozilla/pontoon/blob/main/pontoon/base/templates/widgets/item_selector_list.html#L22.

File: pontoon/settings/base.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (flodolo) [reply pontoon/settings/base.py:424]: I commented on the wrong line 🤦🏼

File: pontoon/administration/forms.py [hunk 3]
```diff
@@ -107,9 +108,24 @@ class Meta:
...
     class Meta:
         model = Tag
-        fields = ("project", "slug", "name", "priority")
+        fields = ("project", "slug", "name", "priority", "resources")
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # If the project instance is available, filter resources for this project
+        if "instance" in kwargs and kwargs["instance"] and kwargs["instance"].project:
```
[the remark this answers is not in the document]
Comment (mathjazz) [reply pontoon/administration/forms.py:124]: Yeah, good point!

File: pontoon/settings/base.py [hunk 4]
```diff
@@ -585,9 +584,9 @@ def _default_from_email():
     "admin_project": {
         "source_filenames": (
             "js/double_list_selector.js",
+            "js/multiple_item_selector.js",
             "js/multiple_team_selector.js",
```
Comment (flodolo) [inline pontoon/settings/base.py:588]: This duplication

Comment (mathjazz) [reply pontoon/settings/base.py:588]: Now, one is CSS, the other is JS.

Comment (flodolo) [reply pontoon/settings/base.py:588]: Apparently my eyes can't see the difference between "item" and "team". Please ignore me ;-)

cant-see-shit

**Comment by bcolsson** [comment]
https://www.youtube.com/watch?v=CnQ8N1KacJc

**Pull request merged**